packages/studio-ui/src/web/views/FlowBuilder/nodeProps/StandardNode.tsx

```diff
@@ -0,0 +1,140 @@
+import { Tag } from '@blueprintjs/core'
+import { lang, Tabs, toast } from 'botpress/shared'
+import { FlowView, NodeView } from 'common/typings'
+import React, { useState } from 'react'
+
+import { sanitizeName } from '~/util'
+
+import EditableInput from '../common/EditableInput'
+
+import ActionSection from './ActionSection'
+import style from './style.scss'
+import TransitionSection from './TransitionSection'
+
+type TabId = 'on-enter' | 'on-receive' | 'transitions'
+
+interface Tab {
+  id: TabId
+  disabled?: boolean
+  className?: string
+  title: string | JSX.Element
+}
+
+interface Props {
+  flow: FlowView
+  node: NodeView
+  subflows: string[]
+  readOnly: boolean
+  isLastNode: boolean
+  updateNode: (node: Partial<NodeView>) => void
+  copyFlowNodeElement: Function
+  pasteFlowNodeElement: Function
+  buffer: any
+}
+
+export default (props: Props) => {
+  const [currentTab, setCurrentTab] = useState<TabId>('on-enter')
+
+  const { node, readOnly, isLastNode } = props
+
+  const tabs: Tab[] = [
+    {
+      id: 'on-enter',
+      title: (
+        <span>
+          <Tag minimal>{node.onEnter?.length ?? 0}</Tag> {lang.tr('studio.flow.node.onEnter')}
+        </span>
+      )
+    },
+    {
+      id: 'on-receive',
+      title: (
+        <span>
+          <Tag minimal>{node.onReceive?.length ?? 0}</Tag> {lang.tr('studio.flow.node.onReceive')}
+        </span>
+      )
+    },
+    {
+      id: 'transitions',
+      title: (
+        <span>
+          <Tag minimal>{node.next?.length ?? 0}</Tag> {lang.tr('studio.flow.node.transitions')}
+        </span>
+      )
+    }
+  ]
+
+  const onChange = (text: string) => {
+    if (!text) {
+      return toast.failure(lang.tr('studio.flow.node.emptyName'))
+    }
+
+    if (text === props.node.name) {
+      return props.node
+    }
+
+    const alreadyExists = props.flow.nodes.find((x) => x.name === text)
+    if (alreadyExists) {
+      return toast.failure(lang.tr('studio.flow.node.nameAlreadyExists'))
+    }
+
+    props.updateNode({ name: text })
+  }
+
+  return (
+    <div className={style.node}>
+      <EditableInput
+        /* We should always sugest that the name should be changed
+             if the node has the default name and it is the last created */
```

Comment:
Is this a TODO?

Reply:
Didn't write this, I think it's a leftover from the old days, I'll leave it there